packages/azure-kusto-data/src/client.ts

```diff
@@ -112,20 +114,43 @@
         if (mappingName != null) {
             endpoint += `&mappingName=${mappingName}`;
         }
+
+        let properties: ClientRequestProperties | null = null;
+        if (clientRequestId) {
+            properties = new ClientRequestProperties();
+            properties.clientRequestId = clientRequestId;
+        }
+
+        return this._execute(endpoint, ExecutionType.Ingest, db, { stream }, properties);
+    }
+
+    async executeStreamingIngestFromBlob(
+        db: string | null,
+        table: string,
+        blob: string,
+        format: any,
+        mappingName: string | null,
+        clientRequestId?: string
+    ): Promise<KustoResponseDataSet> {
+        let endpoint = `${this.endpoints[ExecutionType.Ingest]}/${this.getDb(db)}/${table}?streamFormat=${format}&sourceKind=uri`;
```

Comment:
executeStreamingIngest and executeStreamingIngestFromBlob can be one method with a boolean param I think